src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -39,7 +39,10 @@ sealed class PeerCommand
  * @param maxFeeBasisPoints the max total acceptable fee (all included: service fee and mining fee)
  * @param maxFeeFloor as long as fee is below this amount, it's okay (whatever the percentage is)
  */
-data class RequestChannelOpen(val requestId: ByteVector32, val wallet: WalletState, val maxFeeBasisPoints: Int, val maxFeeFloor: Satoshi) : PeerCommand()
+data class RequestChannelOpen(val requestId: ByteVector32, val wallet: WalletState, val maxFeeBasisPoints: Int, val maxFeeFloor: Satoshi) : PeerCommand() {
+    /** maximum fee that we are willing to pay for this channel */
+    val maxFee = (wallet.confirmedBalance * maxFeeBasisPoints / 10_000).max(maxFeeFloor)
```

Comment:
This could create an issue, you're rounding down in the calculation here, are you sure that eclair doesn't round up?

Reply:
Eclair code is:
```scala
val maxFees: Satoshi = s.tlvStream.get[PleaseOpenChannelTlv.MaxFees] match {
    case Some(maxFees) => (s.localFundingAmount * maxFees.basisPoints / 10_000).max(maxFees.floor) // 100 bips = 1 %
    case None => Long.MaxValue.sat
  }
  if (totalFees > maxFees) {
    log.info(s"rejecting channel request (fee insufficient)")
    sender() ! PleaseOpenChannelRejected(s.requestId, PleaseOpenChannelRejected.Failure.FeeInsufficient, TlvStream(PleaseOpenChannelRejectedTlv.ExpectedFees(totalFees)))
  }
```

So I think it's consistent. But I see your point, we need to test the limit in an integration test.